src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerAssemblyLoader.Core.cs

```diff
@@ -110,6 +111,7 @@ private partial void DisposeWorker()
             {
                 try
                 {
+                    CodeAnalysisEventSource.Log.DisposeAssemblyLoadContext(context.Directory);
```

Comment:
Consider putting this in a finally with a status, or after the actual call to `.Unload` so we don't end up with potentially misleading events.